### PR TITLE
Fixes divide by zero panic

### LINKED ISSUE #268: parser.ParseQuery panics due to integer divide by zero
`parser.ParseQuery` panics with the following input and stack trace.

```
SELECT*FROM t LIMIT-0%.2
```

<details>
<summary>Stack trace</summary>

```
panic: runtime error: integer divide by zero

goroutine 1 [running]:
github.com/genjidb/genji/document.calculateFloats(0x90, 0x55c760, 0x6dfec0, 0xa0, 0x55bf20, 0xc0000a7c60, 0xc0000a7c25, 0x0, 0x0, 0x900000c000120300, ...)
        /home/runner/work/genji/genji/genji/document/value.go:491 +0x846
github.com/genjidb/genji/document.calculateValues(0xc000111c90, 0x55c760, 0x6dfec0, 0xa0, 0x55bf20, 0xc0000a7c60, 0x25, 0x0, 0x90, 0x55c760, ...)
        /home/runner/work/genji/genji/genji/document/value.go:359 +0x30e
github.com/genjidb/genji/document.Value.Mod(...)
        /home/runner/work/genji/genji/genji/document/value.go:324
github.com/genjidb/genji/sql/query/expr.modOp.Eval(0xc000111c80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x40c058, 0x60, ...)
        /home/runner/work/genji/genji/genji/sql/query/expr/arithmeric.go:125 +0x18e
github.com/genjidb/genji/sql/parser.selectConfig.ToTree(0x6a1b14, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/runner/work/genji/genji/genji/sql/parser/select.go:261 +0x4ed
github.com/genjidb/genji/sql/parser.(*Parser).parseSelectStatement(0xc000111c20, 0x55, 0x0, 0x0)
        /home/runner/work/genji/genji/genji/sql/parser/select.go:63 +0x44d
github.com/genjidb/genji/sql/parser.(*Parser).ParseStatement(0xc000111c20, 0x55, 0x0, 0x0, 0x0)
        /home/runner/work/genji/genji/genji/sql/parser/parser.go:91 +0x8a2
github.com/genjidb/genji/sql/parser.(*Parser).ParseQuery(0xc000111c20, 0x5be780, 0xc0000a6000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x18, 0xc000017860)
        /home/runner/work/genji/genji/genji/sql/parser/parser.go:70 +0x214
github.com/genjidb/genji/sql/parser.ParseQuery(0x5be780, 0xc0000a6000, 0xc000017860, 0x18, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc00006de70, ...)
        /home/runner/work/genji/genji/genji/sql/parser/parser.go:41 +0x117
github.com/genjidb/genji/fuzz.FuzzParseQuery(0x7f55a2ef1000, 0x18, 0x18, 0x3)
        /home/runner/work/genji/genji/genji/fuzz/sqlparser.go:13 +0xe9
go-fuzz-dep.Main(0xc00006df70, 0x1, 0x1)
        go-fuzz-dep/main.go:36 +0x1ad
main.main()
        github.com/genjidb/genji/fuzz/go.fuzz.main/main.go:15 +0x52
exit status 2
```
</details>

Post-mortem: when evaluating `x % y` expression, y was rounded to the greatest integer value less than or equal to y. We had an explicit check for y ≠ 0 but not after rounding. That is, y ∈ (0; 1) would panic.

### PR DESCRIPTION
**Pull request by gfelixc**
Closes #268

### CODE REVIEW

File: document/value.go [hunk 1]
```diff
@@ -483,11 +483,12 @@ func calculateFloats(a, b Value, operator byte) (res Value, err error) {
...
 	case '%':
-		if xb == 0 {
+		ia, ib := int64(xa), int64(xb)
+
+		if ib == 0 {
 			return NewNullValue(), nil
 		}
 
-		ia, ib := int64(xa), int64(xb)
 		return NewDoubleValue(float64(ia % ib)), nil
```
Comment (tie) [inline document/value.go:492]: We should use `math.Mod` to compute the remainder of xa/xb.

Special cases (from Go doc):
```
Mod(±Inf, y) = NaN
Mod(NaN, y) = NaN
Mod(x, 0) = NaN
Mod(x, ±Inf) = x
Mod(x, NaN) = NaN
```

**Review by tie — CHANGES_REQUESTED** [review]
@gfelixc thanks for handling this issue! Floating point arithmetic is tricky :)

File: document/value.go (hunk 1 above)
Comment (tie) [reply document/value.go:492]: Special cases are already handled by `math.Mod`, so I think we should just map NaN to NullValue on return.

File: sql/parser/parser_test.go [hunk 2]
```diff
@@ -43,3 +44,9 @@ func TestParserMultiStatement(t *testing.T) {
 		})
 	}
 }
+
+func TestParserDivideByZero(t *testing.T) {
```
Comment (tie) [inline sql/parser/parser_test.go:48]: It’s nice to have a reference for an issue-specific test case.

```suggestion
func TestParserDivideByZero(t *testing.T) {
	// See https://github.com/genjidb/genji/issues/268
```

File: sql/parser/parser_test.go [hunk 3]
```diff
@@ -43,3 +44,9 @@ func TestParserMultiStatement(t *testing.T) {
...
+	require.NotPanics(t, func() {
+		_, _ = ParseQuery(context.Background(), "SELECT*FROM t LIMIT-0%.2")
```
Comment (tie) [inline sql/parser/parser_test.go:50]: A bit more human-readable test case would be
```suggestion
		_, _ = ParseQuery(context.Background(), "SELECT * FROM t LIMIT 0 % .5")
```

File: document/value.go (hunk 1 above)
Comment (gfelixc) [reply document/value.go:492]: You are absolutely right, I focused in solving described issue. WDYT about adding this improvements in a new PR?

Comment (gfelixc) [reply document/value.go:492]: I noticed this PR make no sense since new implementation will overwrite it. I'll do this changes in current PR.

Comment (tie) [reply document/value.go:492]: I think it’s fine to overwrite changes in this PR.

Comment (gfelixc) [reply document/value.go]: @tie done

**Comment by tie** [comment]
@gfelixc we need to update `TestValueMod/double(>_maxint)%int64(100)` test case in `document/value_test.go`.

**Comment by gfelixc** [comment]
> @gfelixc we need to update `TestValueMod/double(>_maxint)%int64(100)` test case in `document/value_test.go`.

@tie done

**Review by tie — APPROVED** [review]

**Review by asdine — APPROVED** [review]
Thanks!

**Pull request merged**